### PR TITLE
add a helper to convert hexadecimal string into nsdata

### PR DESCRIPTION
**Pull request by hyugit**
Hi Kevin,

I updated the helper for hexString conversion. Please take a look.

Thanks,

Huang

### CODE REVIEW

File: BitcoinSwift/NSData+StringEncoding.swift [hunk 1]
```diff
@@ -84,7 +84,32 @@ extension NSData {
...
+    let hexCharToValueMap: [Character: UInt8] = [
+      "0":  0, "1":  1, "2":  2, "3":  3,
+      "4":  4, "5":  5, "6":  6, "7":  7,
+      "8":  8, "9":  9, "a": 10, "b": 11,
+      "c": 12, "d": 13, "e": 14, "f": 15
+    ]
+    let hexChars = Array(hexString.lowercaseString)
+    var byte: UInt8 = 0
+    var bytes: [UInt8] = []
+    for (index, char) in enumerate(hexChars) {
```
Comment (kgreenek) [inline BitcoinSwift/NSData+StringEncoding.swift:99]: nit: just call "index" "i"

Comment (kgreenek) [inline BitcoinSwift/NSData+StringEncoding.swift:99]: This would be a easier to read if you just loop over 2 chars at a time.

Comment (kgreenek) [inline BitcoinSwift/NSData+StringEncoding.swift:99]: Looks good. Just had a couple nits. Please add a unit test :)

Comment (hyugit) [inline BitcoinSwift/NSData+StringEncoding.swift:99]: Great! Thank you for your feedback. Will update it soon

**Comment by trave7er** [comment]
Hi Kevin,

Please take a look at the tests. I'm not sure if I covered all the corner cases.

Thanks,

Huang

File: BitcoinSwiftTests/HexTests.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+//  BitcoinSwift
+//
+//  Created by Huang Yu on 8/26/15.
+//  Copyright (c) 2015 DoubleSha. All rights reserved.
+//
+
+import BitcoinSwift
+import XCTest
+
+class HexTests: XCTestCase {
```
Comment (kgreenek) [inline BitcoinSwiftTests/HexTests.swift:12]: Hmm this seems like it should be combined with what is currently NSDataToUInt8ArrayTests.swift. Maybe rename that file to NSDataTests.swift and combine this into there.

File: BitcoinSwiftTests/HexTests.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    
+    func testHexEncoding() {
+        XCTAssertEqual(hexData.hexString, hexString)
+    }
+    
+    func testHexEncodingWithEmptyData() {
+        XCTAssertEqual(NSData().hexString, "")
+    }
+    
+    func testHexDecoding() {
```
Comment (kgreenek) [inline BitcoinSwiftTests/HexTests.swift:38]: It looks like you're using a tab width of 4 space, but it should be 2. This applies everywhere else too.

File: BitcoinSwiftTests/HexTests.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        168, 215, 146, 122, 74, 61, 211, 81
+    ]
+    
+    var hexData: NSData!
+    
+    let hexString = "51d33d4a7a92d7a85924838c1c278d74b02fe8a443cd35169cc7a581d240ed75"
+    let invalidHexStrings = [ "wronghex", " ", "123%(*&" ]
+    override func setUp() {
+        super.setUp()
+        hexData = NSData(bytes: hexBytes.reverse(), length: hexBytes.count)
```
Comment (kgreenek) [inline BitcoinSwiftTests/HexTests.swift:27]: Why reverse the bytes?

File: BitcoinSwiftTests/HexTests.swift [hunk 5]
```diff
@@ -0,0 +1,59 @@
+//
```
Comment (kgreenek) [inline BitcoinSwiftTests/HexTests.swift:1]: These tests look good. Thank you! :)
Just some nits.
